routes/users.js

```diff
@@ -111,4 +111,6 @@ router.post(
   usersHandler.blockAnonUserFromChat
 );
 
+router.get('/check-allow-dm/:user_id', auth.isAuth, usersHandler.checkAllowAnonDM);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The route `/check-allow-dm/:user_id` is defined with a GET method, which is appropriate for retrieving data. However, it's important to ensure that the `auth.isAuth` middleware and `usersHandler.checkAllowAnonDM` function are correctly implemented to handle potential errors and edge cases. For instance, they should be able to handle situations where the user is not authenticated or the user does not exist.

Also, consider using a more RESTful approach for your endpoint naming. Instead of `/check-allow-dm/:user_id`, you could use something like `/users/:user_id/allow-anonymous-dm`. This makes it clear that we're accessing a property (`allow-anonymous-dm`) of a specific resource (`user`). 

```diff
- router.get('/check-allow-dm/:user_id', auth.isAuth, usersHandler.checkAllowAnonDM);
+ router.get('/users/:user_id/allow-anonymous-dm', auth.isAuth, usersHandler.checkAllowAnonDM);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->